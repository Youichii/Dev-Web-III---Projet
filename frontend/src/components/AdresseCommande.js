import PropTypes from 'prop-types'

const AdresseCommande = ({ className_div, form, Text, id, name, type, placeholder, readOnly }) => {
    if (readOnly) {
        return (
            <div className={className_div}> 
                <label className="label_adresse" for={form}>{Text}</label><br /> 
                <input id={id} name={name} type={type} value={placeholder} readOnly/> 
            </div>
        )
    }
    else {
        return (
            <div className={className_div}> 
                <label className="label_adresse" for={form}>{Text}</label><br /> 
                <input id={id} name={name} type={type} placeholder={placeholder} /> 
            </div>
        )
    }

}

AdresseCommande.propTypes = {
    informations : PropTypes.string,
    type_couleur : PropTypes.string,
    bg_bouton : PropTypes.string,

    className_div : PropTypes.string,
    className_label : PropTypes.string,
    form : PropTypes.string,
    Text : PropTypes.string,
    id : PropTypes.string,
    name : PropTypes.string,
    type : PropTypes.string,
    placeholder : PropTypes.string,
}

export default AdresseCommande