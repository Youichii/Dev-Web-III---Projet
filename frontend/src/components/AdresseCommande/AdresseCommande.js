import PropTypes from 'prop-types'

const AdresseCommande = ({ className_div, className_label, form, Text, id, name, type, placeholder }) => {
    return (
        <div className={className_div}> 
            <label className="label_adresse" htmlFor={form}>{Text}</label><br /> 
            <input id={id} name={name} type={type} placeholder={placeholder} /> 
        </div>
    )

}

/*AdresseCommande.defaultProps = {
    color: 'steelblue',
    text : 'Hello'
}*/

AdresseCommande.propTypes = {
    informations : PropTypes.string,
    type_couleur : PropTypes.string,
    bg_bouton : PropTypes.string,

    className_div : PropTypes.string,
    className_label : PropTypes.string,
    form : PropTypes.string,
    Text : PropTypes.string,
    id : PropTypes.string,
    name : PropTypes.string,
    type : PropTypes.string,
    placeholder : PropTypes.string,
}

export default AdresseCommande
