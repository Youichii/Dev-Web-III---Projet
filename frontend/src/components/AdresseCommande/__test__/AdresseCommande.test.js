import React from 'react';
import ReactDOM from 'react-dom';
import AdresseCommande from '../AdresseCommande';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer  from "react-test-renderer";

//npm test to start all tests
afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<AdresseCommande></AdresseCommande>, div)
})

/*it("render button text correctly", () => {
    const {getByTestId} = render(<AdresseCommande color="green" text='Hello' onClick={()=> {console.log("hello")}} ></AdresseCommande>);
    expect(getByTestId('button')).toHaveTextContent("Hello")
})

it("render button color correctly", () => {
    const {getByTestId} = render(<AdresseCommande color="green" text='Hello' onClick={()=> {console.log("hello")}} ></AdresseCommande>);
    expect(getByTestId('button')).toHaveStyle("backgroundColor : green")
})*/

it("matches snapshot 1", () => {
    const tree = renderer.create(<AdresseCommande className_div="i_adresse_livraison" fom="adresse_livraison" Text="Adresse de livraison" id="adresse_livraison" name="add_livraison" type="text" placeholder="Rue du paradis"></AdresseCommande>).toJSON(); //convert to a virtualDOM object
    expect(tree).toMatchSnapshot(); 
}) 

it("matches snapshot 2", () => {
    const tree = renderer.create(<AdresseCommande className_div="i_adresse_livraison" fom="adresse_livraison" Text="Adresse de livraison" id="adresse_livraison" name="add_livraison" type="text" placeholder="Rue du paradis"></AdresseCommande>).toJSON(); //convert to a virtualDOM object
    expect(tree).toMatchSnapshot();
}) 