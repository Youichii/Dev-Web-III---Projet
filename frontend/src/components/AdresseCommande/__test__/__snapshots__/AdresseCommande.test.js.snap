// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 AdresseCommande 1`] = `
<div
  className="i_adresse_livraison"
  data-testid="adresse_commande"
>
  <label
    className="label_adresse"
  >
    Adresse de livraison
  </label>
  <br />
  <input
    id="adresse_livraison"
    name="add_livraison"
    placeholder="Rue du paradis"
    type="text"
  />
</div>
`;
